app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -319,10 +320,30 @@ class BrowserTabViewModel(
         browserViewState.value = currentBrowserViewState().copy(isFireproofWebsite = isFireproofWebsite())
     }
 
+    @ExperimentalCoroutinesApi
     private val fireButtonAnimation = Observer<Boolean> { shouldShowAnimation ->
+        Timber.i("shouldShowAnimation $shouldShowAnimation")
         if (currentBrowserViewState().fireButton is FireButton.Visible) {
             browserViewState.value = currentBrowserViewState().copy(fireButton = FireButton.Visible(pulseAnimation = shouldShowAnimation))
         }
+
+        if (shouldShowAnimation) {
+            registerAndScheduleDismissAction()
+        }
+    }
+
+    @ExperimentalCoroutinesApi
+    private fun registerAndScheduleDismissAction() {
```

Comment:
Haven't check the feasibility of this but it would be nice to test this new logic and make sure `dismissPulseAnimation()` gets called after `On_HOUR_IN_MS`, etc. Again, not sure how easy this would be to test. Easy route would be to not make the function private but don't do that :D.

Reply:
Agree, but as you said it's not straightforward. I created this task inside the project in order to add some tests before we do the final release (winner variant).
https://app.asana.com/0/0/1198183818260594/f